### DISCUSSION TITLE
Unable to delete user from authentication

### DISCUSSION THREAD

**Discussion by josh4500** [supabase/supabase#3284] [Questions] [answered]
# Bug report

## Describe the bug

Error Deleting user failed: update or delete on table "users" violates foreign key constraint "objects_owner_fkey" on table "objects"

I have a table "users" which I can't delete and didn't use any of the value from Authentication.

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Go to 'Authentication'
2. Click on 'more (...)'
3.  Click on ' Delete user'

## Expected behavior
User should be removed from the table.

## Screenshots
Authentication 
![Screenshot 2021-09-22 090840](https://user-images.githubusercontent.com/61246632/134306590-bec694db-2f8b-4c55-8767-a26cf7a2ab0f.jpg)

User table
![Screenshot 2021-09-22 091118](https://user-images.githubusercontent.com/61246632/134306923-db9c7eb9-f1ec-4303-b8d1-732927333b39.jpg)

## System information

- Browser : Chrome

**Comment by bnjmnt4n** [comment]
See https://github.com/supabase/storage-api/issues/65.

**Comment by kiwicopple** [comment, marked as answer]
Hi @bnjmnt4n - it looks like you are "referencing" the user from another place in your database - maybe in one of your `public` tables. You would need to delete that data first before you can remove the user, or you can use [cascade deletes](https://stackoverflow.com/questions/11504699/postgresql-on-delete-cascade).

This is less of a bug, and more of a Postgres functionality. I'll move this to a discussion!

**Comment by RobLeachman** [comment]
I'm hitting the same error today. Just getting started, delete all the rows on my tables so ... it is a puzzle.

**Comment by RobLeachman** [comment]
Hmm OK it was Storage, the avatar. The instructions are clear enough I just didn't get to the part where we add triggers to clean up storage as the authentication user record is deleted. No bigs.

**Reply by SrGeneroso** [reply]
I have a couple of rows also in storage, how did you delete those?
